tests/FormlyMapper/FormlyField/BirthdayFieldTest.php

```diff
@@ -0,0 +1,287 @@
+        ];
+    }
+
+    /**
+     * @dataProvider basicDataProvider
+     */
+    public function testIsAddingBirthdayFields(array $fieldConfiguration, array $expected): void
+    {
+        $fieldConfiguration['validation'] = [
+            'Symfony\Component\Validator\Constraints\Regex' => [
+                'pattern' => '^[0-9]{4}\-[0-9]{2}\-[0-9]{2}$',
+                'message' => 'The birthday field must follow the pattern "yyyy-MM-dd"',
+            ],
+        ];
+
+        $expected['templateOptions']['pattern'] = '^[0-9]{4}\-[0-9]{2}\-[0-9]{2}$';
+        $expected['validation'] = [
+            'messages' => [
+                'regex' => 'The birthday field must follow the pattern "yyyy-MM-dd"',
+            ],
+        ];
+
+        $this->formlyField->setFieldConfiguration($fieldConfiguration);
+        $actual = $this->formlyField->getFormlyFieldConfiguration();
+
+        $this->assertEquals($expected, $actual);
+    }
+
+    /**
+     * @dataProvider basicDataProvider
+     */
+    public function testIsAddingSpecificYears(array $fieldConfiguration, array $expected): void
+    {
+        $years = [2000, 2001, 2002, 2003, 2004, 2005];
+
+        $fieldConfiguration['options']['years'] = $years;
+
+        $expected['templateOptions']['years'] = $years;
+
+        $this->formlyField->setFieldConfiguration($fieldConfiguration);
+        $actual = $this->formlyField->getFormlyFieldConfiguration();
+
+        $this->assertEquals($expected, $actual);
+    }
+
+    /**
+     * @dataProvider basicDataProvider
+     */
+    public function testWithNotNumberSpecificYears(array $fieldConfiguration, array $expected): void
+    {
+        $goodYears = [2000, 2001, 2002, 2003, 2004, 2005];
+        $badYears = ['2000', 2001, '2002', 2003, 2004, 2005];
+
+        $fieldConfiguration['options']['years'] = $badYears;
+
+        $expected['templateOptions']['years'] = $goodYears;
+
+        $this->formlyField->setFieldConfiguration($fieldConfiguration);
+        $actual = $this->formlyField->getFormlyFieldConfiguration();
+
+        $this->assertEquals($expected, $actual);
+    }
+
+    /**
+     * @dataProvider basicDataProvider
+     */
+    public function testIsAddingYearsRangeInOrderByDefault(array $fieldConfiguration, array $expected): void
+    {
+        $minAgeAllowed = 18;
+        $maxAgeAllowed = 120;
+
+        $fieldConfiguration['options']['minAgeAllowed'] = $minAgeAllowed;
+        $fieldConfiguration['options']['maxAgeAllowed'] = $maxAgeAllowed;
+
+        $expected['templateOptions']['years'] = range(date('Y') - $minAgeAllowed, date('Y') - $maxAgeAllowed);
+
+        $this->formlyField->setFieldConfiguration($fieldConfiguration);
+        $actual = $this->formlyField->getFormlyFieldConfiguration();
+
+        $this->assertEquals($expected, $actual);
+    }
+
+    /**
+     * @dataProvider basicDataProvider
+     */
+    public function testWithBadAgeRangeForWord(array $fieldConfiguration, array $expected): void
```

Comment:
What is the test actually trying to accomplish? It looks like you are trying to bypass the functionality when the configuration contains a "word" instead of a number. The test name ending in `ForWord` doesn't really make a lot of sense. Instead, a better test name would be `testItDoesNotAllowNonNumericalValuesForAllowedAges` or something of the sort.

Reply:
fixed, thanks for the correction!